UserDataDeleteOrchestrator/handler.ts

```diff
@@ -0,0 +1,184 @@
+  context: IFunctionContext
+): IterableIterator<unknown> {
+  const document = context.df.getInput();
+  // This check has been done on the trigger, so it should never fail.
+  // However, it's worth the effort to check it twice
+  const invalidInputOrCurrentUserDataProcessing = ProcessableUserDataDelete.decode(
+    document
+  ).mapLeft<InvalidInputFailure>(err => {
+    context.log.error(
+      `${logPrefix}|WARN|Cannot decode ProcessableUserDataDelete document: ${readableReport(
+        err
+      )}`
+    );
+    return InvalidInputFailure.encode({
+      kind: "INVALID_INPUT",
+      reason: readableReport(err)
+    });
+  });
+
+  if (isLeft(invalidInputOrCurrentUserDataProcessing)) {
+    return invalidInputOrCurrentUserDataProcessing.value;
+  }
+
+  const currentUserDataProcessing =
+    invalidInputOrCurrentUserDataProcessing.value;
+
+  try {
+    // lock user session
+    SetUserSessionLockActivityResultSuccess.decode(
+      yield context.df.callActivity(
+        "SetUserSessionLockActivity",
+        SetUserSessionLockActivityInput.encode({
+          action: "LOCK",
+          fiscalCode: currentUserDataProcessing.fiscalCode
+        })
+      )
+    ).getOrElseL(err => {
+      throw toActivityFailure(err, "SetUserSessionLockActivity", {
+        action: "LOCK"
+      });
+    });
+
+    // set as wip
+    SetUserDataProcessingStatusActivityResultSuccess.decode(
+      yield context.df.callActivity("SetUserDataProcessingStatusActivity", {
+        currentRecord: currentUserDataProcessing,
+        nextStatus: UserDataProcessingStatusEnum.WIP
+      })
+    ).getOrElseL(err => {
+      throw toActivityFailure(err, "SetUserDataProcessingStatusActivity", {
+        status: UserDataProcessingStatusEnum.WIP
+      });
+    });
+
+    //
+    // TODO: delete data
+    //
+
+    // unlock user
+    SetUserSessionLockActivityResultSuccess.decode(
+      yield context.df.callActivity(
+        "SetUserSessionLockActivity",
+        SetUserSessionLockActivityInput.encode({
+          action: "UNLOCK",
+          fiscalCode: currentUserDataProcessing.fiscalCode
+        })
+      )
+      // tslint:disable-next-line: no-identical-functions
+    ).getOrElseL(err => {
+      throw toActivityFailure(err, "SetUserSessionLockActivity", {
+        action: "UNLOCK"
+      });
+    });
+
+    // set as closed
+    SetUserDataProcessingStatusActivityResultSuccess.decode(
+      yield context.df.callActivity("SetUserDataProcessingStatusActivity", {
+        currentRecord: currentUserDataProcessing,
+        nextStatus: UserDataProcessingStatusEnum.CLOSED
+      })
+    ).getOrElseL(err => {
+      throw toActivityFailure(err, "SetUserDataProcessingStatusActivity", {
+        status: UserDataProcessingStatusEnum.CLOSED
+      });
+    });
+    return OrchestratorSuccess.encode({ kind: "SUCCESS" });
+  } catch (error) {
+    context.log.error(
+      `${logPrefix}|ERROR|Failed processing user data for download: ${error.message}`
```

Comment:
error is any here (not an instance of Error) so may not have a `message` property